packages/react-core/src/components/Panel/PanelMain.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Panel/panel';
+import { css } from '@patternfly/react-styles';
+
+export interface PanelMainProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered inside the panel main div */
+  children?: React.ReactNode;
+  /** Class to add to outer div */
+  className?: string;
+  /** Max height of the panel main div as a number of px or string percentage */
+  maxHeight?: string | number;
+}
+
+export const PanelMain: React.FunctionComponent<PanelMainProps> = ({
+  className,
+  children,
+  maxHeight,
+  ...props
+}: PanelMainProps) => {
+  const compiledProps = {
+    className: css(styles.panelMain, className),
+    ...(maxHeight && { style: { maxHeight, overflow: 'auto' } })
```

Comment:
Here's a better example than the select for how to use the CSS vars available to set this `maxHeight` value. Also you shouldn't need to specify `overflow` - the user should be using `isScrollable`, which will set `overflow` in the CSS.

https://github.com/patternfly/patternfly-react/blob/8c1eb5d895aa2512f7b4197654e414394ff25b30/packages/react-core/src/components/Menu/MenuContent.tsx#L32-L44

The var to set for the panel max height is `--pf-c-panel__main--MaxHeight`